File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/Integration/Movie/MovieRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace UserControlForm.Integration;

[ConnectionKey("Default"), Module("Integration"), TableName("Movie")]
[DisplayName("Movie"), InstanceName("Movie")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
public sealed class MovieRow : Row<MovieRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Name"), Size(50), QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    [DisplayName("Type")]
    public int? Type { get => fields.Type[this]; set => fields.Type[this] = value; }

    [DisplayName("Description"), Size(50)]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField Name;
        public Int32Field Type;
        public StringField Description;

    }
}

File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/ServerTypes/Integration/MovieRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface MovieRow {
    Id?: number;
    Name?: string;
    Type?: number;
    Description?: string;
}

export abstract class MovieRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Integration.Movie';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<MovieRow>();
}